sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file    ng_netreg.c
+ */
+
+#include <errno.h>
+
+#include "clist.h"
+#include "net/ng_netreg.h"
+#include "net/ng_nettype.h"
+
+#define _NETREG_PROTO_REG_SIZE  (NG_NETTYPE_NUMOF)
+
+static ng_netreg_entry_t *netreg[NG_NETTYPE_NUMOF];
+static ng_netreg_entry_t netreg_entries[NG_NETREG_SIZE];
+
+int ng_netreg_register(ng_nettype_t type, kernel_pid_t proto_pid, uint32_t demux_ctx,
+                       kernel_pid_t target_pid)
+{
+    ng_netreg_entry_t *entry = NULL;
+
+    for (int i = 0; i < NG_NETREG_SIZE; i++) {
+        if (netreg_entries[i].next == NULL) {
+            entry = &(netreg_entries[i]);
```

Comment:
hm, in the beginning all `netreg_entries[]` elements are null, so you always return `ENOMEM`?!